src/cudamatrix/cu-math.cc

```diff
@@ -226,8 +227,19 @@ void ComputeXvectorObjfFromScores(const CuMatrixBase<Real> &scores,
   } else
   #endif
   {
-    // TODO: Add the CPU version.
-    KALDI_LOG << "NOT USING CUDA";
+    int32 num_rows = scores.NumRows();
+    BaseFloat K = 1.0 / (num_rows - 2.0);
+    for (int32 i = 0; i < num_rows; i++) {
+      for (int32 j = i + 1; j < num_rows; j++) {
+        if (i + 1 == j && i % 2 == 0) {
+          (*objf_terms)(i, j) = log(1.0 + exp(-scores(i, j)));
```

Comment:
you'll need to be a little more careful about the possibility of overflow.  I suggest using something like
  x > 15.0 ? x : log(1.0 + exp(-x))